fs_attachment/models/ir_attachment.py

```diff
@@ -456,7 +457,7 @@ def _enforce_meaningful_storage_filename(self) -> None:
             # we need to update the store_fname with the new filename by
             # calling the write method of the field since the write method
             # of ir_attachment prevent normal write on store_fname
-            attachment._force_write_store_fname(f"{storage}://{new_filename}")
+            attachment._force_write_store_fname(f"{storage}://{new_filename_with_path}")
```

Comment:
@wpichler The name should not contains the directory path defined on the backend.

Reply:
If you nest your filesystem into 'rooted_dir' filesystem, the path given as parameter to the `rooted_dir` must not appear into the store_fname.  (When you define a `directory_path` on your backend,
your filesystem is nested into a `rooted_dir` filesystem. This nesting could be done on multiple levels)

Reply:
I think we can manage-it by determining which part of the path is related to the nested rooted_dir filesystem and only remove this part of the path from the new_filename_with_path.... A `rooted_dir` has a path attribute and a fs_attribute...

something like.....

```python

# rooted_dir path computation...
rooted_paths = []
rooted_dir_fs = fs

while rooted_dir_fs and hasattr(rooted_dir_fs, "path"):
    rooted_paths.append(rooted_dir_fs.path)
    rooted_dir_fs = getattr(rooted_dir_fs, "fs", None")
    
rooted_paths.reverse()
rooted_path = "/".join(rooted_paths)

new_filename = new_filename_with_path.replace(rooted_path, "")

```

In any cases, we must add unit tests for such a cases.

Reply:
I've applied your code (with comments that code is coming from your side....) - tested it - does work fine for me.

Reply:
We need unit tests. IMO, the code could be extracted into a model method  on fs.storage.